src/main/java/com/amazon/jenkins/ec2fleet/CloudNames.java

```diff
@@ -1,13 +1,19 @@
 package com.amazon.jenkins.ec2fleet;
 
 import jenkins.model.Jenkins;
+import org.apache.commons.lang.RandomStringUtils;
 
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class CloudNames {
 
+  public static final int SUFFIX_LENGTH = 8;
+
+  public static Set<String> usedSuffixes = new HashSet<String>();
```

Comment:
We could potentially do without this change since the probability of a repeat cloud name is $\frac{x}{62^{y}}$ where x is the total number of clouds constructed and y is `SUFFIX_LENGTH`. Even if x is 1 billion thats only a 0.0004% chance of getting a collision.